base/task.jl

```diff
@@ -992,3 +992,17 @@ if Sys.iswindows()
 else
     pause() = ccall(:pause, Cvoid, ())
 end
+
+interrupt_handlers() = ccall(:jl_get_interrupt_handlers, Any, ())::Vector{Task}
+function register_interrupt_handler(t::Task)
+    handlers = interrupt_handlers()
+    if findfirst(==(t), handlers) === nothing
+        push!(handlers, t)
```

Comment:
This is thead-unsafe, so we should probably think about whether to do all of this in C with a lock, instead of half here and half in C with multiple locks

Reply:
Moved this logic into C with a `jl_mutex_t`